src/com/github/korniloval/matlab/editor/smartEnter/StatementCompletionProcessor.kt

```diff
@@ -0,0 +1,64 @@
+package com.github.korniloval.matlab.editor.smartEnter
+
+import com.github.korniloval.matlab.psi.MatlabAssignExpr
+import com.github.korniloval.matlab.psi.MatlabFunctionExpr
+import com.intellij.codeInsight.editorActions.smartEnter.SmartEnterProcessor
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import java.util.*
+import java.util.Optional.*
+
+class StatementCompletionProcessor : SmartEnterProcessor() {
+
+  override fun process(project: Project, editor: Editor, psiFile: PsiFile): Boolean {
+    return ofNullable(getStatementAtCaret(editor, psiFile))
+        .flatMap { element ->
+          findStatement(element) { node ->
+            when (node) {
+              is MatlabFunctionExpr,
+              is MatlabAssignExpr -> true
+              else -> false
+            }
+          }
+        }
+        .map { statementElement ->
+          val statementRange = getStatementRange(editor, statementElement)
+          val fullStatement = editor.document.getText(statementRange)
+          val endOfStatement =
+              completeStatementIfNecessary(fullStatement, editor, statementElement.textRange.endOffset)
+          editor.caretModel.moveToOffset(endOfStatement + 1)
+          true
+        }.orElse(true)
```

Comment:
Seems like the function always returns `true`, am I right? It should return `true` only if it completed the statement.

Kotlin has nullable types that should be used instead of `Optional`. It will be idiomatic and more understandable for Java people who are unfamiliar with semantics of `map` and `flatMap` of `Optional`.

So I suggest re-writing this function without using `Optional` like this (or similar):
```kt
val element = getStatementAtCaret(editor, psiFile) ?: return false
val statement = findStatement(element) { node -> // findStatement returns `PsiElement?`
    when (node) {
        is MatlabFunctionExpr,
        is MatlabAssignExpr -> true
        else -> false
    }
}
if (statement == null) return false
val statementRange = ...
...
return true
```

Reply:
Elvis operator inspection sometimes annoys me and I just ignore it :) but it's okay to use the operator here